widget/textgrid.go

```diff
@@ -444,65 +703,65 @@ func (t *textGridRenderer) addCellsIfRequired() {
-			pad := t.lineNumberWidth() - len(lineStr)
-			for ; i < pad; i++ {
-				t.setCellRune(' ', x, TextGridStyleWhitespace, rowStyle) // padding space
-				x++
-			}
-			for c := 0; c < len(lineStr); c++ {
-				t.setCellRune(lineStr[c], x, TextGridStyleDefault, rowStyle) // line numbers
-				i++
-				x++
-			}
-
-			t.setCellRune('|', x, TextGridStyleWhitespace, rowStyle) // last space
+	row := t.obj.text.text.Rows[t.obj.row]
+	rowStyle := row.Style
+	i := 0
+	if t.obj.text.text.ShowLineNumbers {
+		lineStr := []rune(strconv.Itoa(t.obj.row + 1))
+		pad := t.lineNumberWidth() - len(lineStr)
+		for ; i < pad; i++ {
+			t.setCellRune(' ', x, TextGridStyleWhitespace, rowStyle) // padding space
+			x++
+		}
+		for c := 0; c < len(lineStr); c++ {
+			t.setCellRune(lineStr[c], x, TextGridStyleDefault, rowStyle) // line numbers
 			i++
 			x++
 		}
-		for _, r := range row.Cells {
-			if i >= t.cols { // would be an overflow - bad
-				continue
+
+		t.setCellRune('|', x, TextGridStyleWhitespace, rowStyle) // last space
+		i++
+		x++
+	}
+	for _, r := range row.Cells {
+		if i >= t.cols { // would be an overflow - bad
+			continue
+		}
+		if t.obj.text.text.ShowWhitespace && (r.Rune == ' ' || r.Rune == '\t') {
+			sym := textAreaSpaceSymbol
+			if r.Rune == '\t' {
+				sym = textAreaTabSymbol
 			}
-			if t.text.ShowWhitespace && (r.Rune == ' ' || r.Rune == '\t') {
-				sym := textAreaSpaceSymbol
-				if r.Rune == '\t' {
-					sym = textAreaTabSymbol
-				}
 
-				if r.Style != nil && r.Style.BackgroundColor() != nil {
-					whitespaceBG := &CustomTextGridStyle{FGColor: TextGridStyleWhitespace.TextColor(),
-						BGColor: r.Style.BackgroundColor()}
-					t.setCellRune(sym, x, whitespaceBG, rowStyle) // whitespace char
-				} else {
-					t.setCellRune(sym, x, TextGridStyleWhitespace, rowStyle) // whitespace char
-				}
+			if r.Style != nil && r.Style.BackgroundColor() != nil {
+				whitespaceBG := &CustomTextGridStyle{FGColor: TextGridStyleWhitespace.TextColor(),
+					BGColor: r.Style.BackgroundColor()}
+				t.setCellRune(sym, x, whitespaceBG, rowStyle) // whitespace char
 			} else {
-				t.setCellRune(r.Rune, x, r.Style, rowStyle) // regular char
+				t.setCellRune(sym, x, TextGridStyleWhitespace, rowStyle) // whitespace char
 			}
-			i++
-			x++
-		}
-		if t.text.ShowWhitespace && i < t.cols && rowIndex < len(t.text.Rows)-1 {
-			t.setCellRune(textAreaNewLineSymbol, x, TextGridStyleWhitespace, rowStyle) // newline
-			i++
-			x++
-		}
-		for ; i < t.cols; i++ {
-			t.setCellRune(' ', x, TextGridStyleDefault, rowStyle) // blanks
-			x++
+		} else {
+			t.setCellRune(r.Rune, x, r.Style, rowStyle) // regular char
 		}
-
-		line++
+		i++
+		x++
+	}
+	if t.obj.text.text.ShowWhitespace && i < t.cols && t.obj.row < len(t.obj.text.text.Rows)-1 {
+		t.setCellRune(textAreaNewLineSymbol, x, TextGridStyleWhitespace, rowStyle) // newline
+		i++
+		x++
+	}
+	for ; i < t.cols; i++ {
+		t.setCellRune(' ', x, TextGridStyleDefault, rowStyle) // blanks
+		x++
```

Comment:
I'm not sure if there is much to do about this but I find that the `x++`, `i++` and so on everywhere makes the code rather hard to read.

Reply:
All I did was un-indent the existing code here by working on a single row instead of them all

Reply:
Ah, sorry. That was not apparent from the diff but I switched to side-by-side and that's more clear